gpjax/types.py

```diff
@@ -7,17 +7,20 @@
 NoneType = type(None)
 Array = tp.Union[np.ndarray, jnp.ndarray]
 
+import tensorflow.data as tfd
 
-@dataclass(repr=False)
+
+@dataclass
 class Dataset:
+    """GPJax Dataset."""
+
     X: Array
     y: Array = None
+    _tf_dataset: tp.Optional[tfd.Dataset] = None
```

Comment:
Perhaps acknowledge what this is in the object's docstring e.g., line 15. It's not immediately clear

Reply:
Absolutely -> see https://github.com/thomaspinder/GPJax/pull/48/commits/64364a4fa3fbbbec39262f733017a4af6edd1648.